rdbms/src/main/java/org/spine3/server/storage/jdbc/projection/query/ProjectionStorageQueryFactory.java

```diff
@@ -0,0 +1,64 @@
+package org.spine3.server.storage.jdbc.projection.query;
+
+
+import com.google.protobuf.Timestamp;
+import org.slf4j.Logger;
+import org.spine3.server.entity.Entity;
+import org.spine3.server.storage.jdbc.util.DataSourceWrapper;
+import org.spine3.server.storage.jdbc.util.IdColumn;
+
+import static org.spine3.server.storage.jdbc.projection.query.Constants.LAST_EVENT_TIME_TABLE_NAME_SUFFIX;
+import static org.spine3.server.storage.jdbc.util.DbTableNameFactory.newTableName;
+
+public class ProjectionStorageQueryFactory<I> {
+
+    private final IdColumn<I> idColumn;
+    private final String tableName;
+    private final DataSourceWrapper dataSource;
+    private Logger logger;
+
+    public ProjectionStorageQueryFactory(DataSourceWrapper dataSource, Class<? extends Entity<I, ?>> projectionClass) {
+        this.idColumn = IdColumn.newInstance(projectionClass);
+        this.tableName = newTableName(projectionClass) + LAST_EVENT_TIME_TABLE_NAME_SUFFIX;
+        this.dataSource = dataSource;
+    }
+
+    public void setLogger(Logger logger) {
+        this.logger = logger;
+    }
+
+    public CreateTableQuery newCreateTableQuery() {
+        final CreateTableQuery.Builder builder = CreateTableQuery.newBuilder()
+                .setDataSource(dataSource)
+                .setLogger(logger)
+                .setTableName(tableName);
+
```

Comment:
Pls read Google code conventions. They do not add empty lines (so do we). But we can add empty lines in test methods to separate given, when , then.